src/statsbase.jl

```diff
@@ -0,0 +1,186 @@
+using StatsBase
+
+
+# Support some of the weighted statistics function in StatsBase
+# NOTES:
+# - Ambiguity errors are still possible for weights with overly specific methods (e.g., UnitWeights)
+# - Ideally, when the weighted statistics is moved to Statistics.jl we can remove this entire file.
+#   https://github.com/JuliaLang/Statistics.jl/pull/2
+function Statistics.mean(A::KeyedArray, wv::AbstractWeights; dims=:, kwargs...)
+    dims === Colon() && return mean(parent(A), wv; kwargs...)
+    numerical_dims = AxisKeys.hasnames(A) ? NamedDims.dim(dimnames(A), dims) : dims
+    data = mean(parent(A), wv; dims=numerical_dims, kwargs...)
+    new_keys = ntuple(d -> d in numerical_dims ? Base.OneTo(1) : axiskeys(A,d), ndims(A))
+    return KeyedArray(data, map(copy, new_keys))#, copy(A.meta))
+end
+
+function Statistics.mean(A::KeyedArray, wv::KeyedVector; kwargs...)
+    return mean(A, hasnames(wv) ? parent(parent(wv)) : parent(wv); kwargs...)
+end
+
+# var and std are separate cause they don't use the dims keyword and we need to set corrected=true
+for fun in [:var, :std]
+    @eval function Statistics.$fun(A::KeyedArray, wv::AbstractWeights; dims=:, corrected=true, kwargs...)
+        dims === Colon() && return $fun(parent(A), wv; kwargs...)
+        numerical_dims = AxisKeys.hasnames(A) ? NamedDims.dim(dimnames(A), dims) : dims
+        data = $fun(parent(A), wv, numerical_dims; corrected=corrected, kwargs...)
+        new_keys = ntuple(d -> d in numerical_dims ? Base.OneTo(1) : axiskeys(A,d), ndims(A))
+        return KeyedArray(data, map(copy, new_keys))#, copy(A.meta))
+    end
+
+    @eval function Statistics.$fun(A::KeyedArray, wv::KeyedVector; kwargs...)
+        return $fun(A, hasnames(wv) ? parent(parent(wv)) : parent(wv); kwargs...)
+    end
+end
+
+function Statistics.cov(A::KeyedMatrix, wv::AbstractWeights; dims=1, corrected=true, kwargs...)
```

Comment:
Can this be combined with `Statistics.cor` in an `@eval` loop? It looks like the only difference is the keyword `corrected=true`, but won't this be passed along as part of `kwargs...` anyway?

Reply:
Yeah, it's just that the default behaviour in StatsBase will throw a deprecation warning stating that `corrected=true` will be the default in a future release. If we're okay with the deprecation in our tests then I can merge them and the warning will go away in a future release.

Reply:
Oh I see. I guess I don't mind the noise, and eventually it will stop as you say.

Reply:
N/m, just changed the tests to explicitly pass `corrected=true` to avoid the noise.